core/trino-main/src/test/java/io/trino/operator/scalar/TestDateTimeFunctions.java

```diff
@@ -309,6 +310,7 @@ public void testDate()
     @Test
     public void testFromISO8601()
     {
+        assertFunction("from_iso8601_timestamp('" + TIMESTAMP_ISO8601_BASIC_STRING + "')", TIMESTAMP_WITH_TIME_ZONE, toTimestampWithTimeZone(TIMESTAMP_WITH_NUMERICAL_ZONE));
```

Comment:
Added test is failing. 
```
Error:  io.trino.operator.scalar.TestDateTimeFunctions.testFromISO8601  Time elapsed: 0.036 s  <<< FAILURE!
io.trino.spi.TrinoException: Invalid format: "20010822T030405.321-1100" is malformed at "0405.321-1100"
	at io.trino.operator.scalar.DateTimeFunctions.parseDateTimeHelper(DateTimeFunctions.java:356)
	at io.trino.operator.scalar.DateTimeFunctions.fromISO8601Timestamp(DateTimeFunctions.java:215)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:732)
	at java.base/java.lang.invoke.MethodHandle.invokeWithArguments(MethodHandle.java:757)
	at io.trino.sql.InterpretedFunctionInvoker.invoke(InterpretedFunctionInvoker.java:101)
	at io.trino.sql.planner.ExpressionInterpreter$Visitor.visitFunctionCall(ExpressionInterpreter.java:1051)
	at io.trino.sql.tree.FunctionCall.accept(FunctionCall.java:121)
	at io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at io.trino.sql.planner.ExpressionInterpreter$Visitor.processWithExceptionHandling(ExpressionInterpreter.java:305)
	at io.trino.sql.planner.ExpressionInterpreter$Visitor.visitRow(ExpressionInterpreter.java:1313)
	at io.trino.sql.tree.Row.accept(Row.java:54)
	at io.trino.sql.tree.AstVisitor.process(AstVisitor.java:27)
	at io.trino.sql.planner.ExpressionInterpreter$Visitor.processWithExceptionHandling(ExpressionInterpreter.java:305)
	at io.trino.sql.planner.ExpressionInterpreter.evaluate(ExpressionInterpreter.java:271)
	at io.trino.sql.planner.LocalExecutionPlanner$Visitor.visitValues(LocalExecutionPlanner.java:2046)
	at io.trino.sql.planner.LocalExecutionPlanner$Visitor.visitValues(LocalExecutionPlanner.java:846)
	at io.trino.sql.planner.plan.ValuesNode.accept(ValuesNode.java:131)
	at io.trino.sql.planner.LocalExecutionPlanner$Visitor.visitOutput(LocalExecutionPlanner.java:934)
	at io.trino.sql.planner.LocalExecutionPlanner$Visitor.visitOutput(LocalExecutionPlanner.java:846)
	at io.trino.sql.planner.plan.OutputNode.accept(OutputNode.java:83)
	at io.trino.sql.planner.LocalExecutionPlanner.plan(LocalExecutionPlanner.java:568)
	at io.trino.testing.LocalQueryRunner.createDrivers(LocalQueryRunner.java:957)
	at io.trino.testing.LocalQueryRunner.executeInternal(LocalQueryRunner.java:835)
	at io.trino.testing.LocalQueryRunner.lambda$executeWithPlan$6(LocalQueryRunner.java:803)
	at io.trino.transaction.TransactionBuilder.execute(TransactionBuilder.java:150)
	at io.trino.testing.LocalQueryRunner.inTransaction(LocalQueryRunner.java:815)
	at io.trino.testing.LocalQueryRunner.executeWithPlan(LocalQueryRunner.java:803)
	at io.trino.testing.LocalQueryRunner.execute(LocalQueryRunner.java:797)
	at io.trino.testing.LocalQueryRunner.execute(LocalQueryRunner.java:791)
	at io.trino.operator.scalar.FunctionAssertions.executeProjectionWithAllInTx(FunctionAssertions.java:563)
	at io.trino.operator.scalar.FunctionAssertions.lambda$executeProjectionWithAll$10(FunctionAssertions.java:548)
	at io.trino.transaction.TransactionBuilder.execute(TransactionBuilder.java:150)
	at io.trino.operator.scalar.FunctionAssertions.executeProjectionWithAll(FunctionAssertions.java:545)
	at io.trino.operator.scalar.FunctionAssertions.selectUniqueValue(FunctionAssertions.java:342)
	at io.trino.operator.scalar.FunctionAssertions.selectSingleValue(FunctionAssertions.java:337)
	at io.trino.operator.scalar.FunctionAssertions.assertFunction(FunctionAssertions.java:300)
	at io.trino.operator.scalar.AbstractTestFunctions.assertFunction(AbstractTestFunctions.java:90)
	at io.trino.operator.scalar.TestDateTimeFunctions.testFromISO8601(TestDateTimeFunctions.java:313)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:77)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:568)
	at org.testng.internal.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:104)
	at org.testng.internal.Invoker.invokeMethod(Invoker.java:645)
	at org.testng.internal.Invoker.invokeTestMethod(Invoker.java:851)
	at org.testng.internal.Invoker.invokeTestMethods(Invoker.java:1177)
	at org.testng.internal.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:129)
	at org.testng.internal.TestMethodWorker.run(TestMethodWorker.java:112)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1136)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:635)
	at java.base/java.lang.Thread.run(Thread.java:833)
Caused by: java.lang.IllegalArgumentException: Invalid format: "20010822T030405.321-1100" is malformed at "0405.321-1100"
	at org.joda.time.format.DateTimeFormatter.parseDateTime(DateTimeFormatter.java:945)
	at io.trino.operator.scalar.DateTimeFunctions.parseDateTimeHelper(DateTimeFunctions.java:353)
	... 50 more
```

Reply:
> Added test is failing. Both `from_iso8601_timestamp` & `from_iso8601_timestamp_nanos` support only extended format as far as I know.

Yes, that was the idea. Keeping the test failing until somebody make the implementation. 

Test Driven Development ;-)

Unfortunately I don't even remember which was the software that I was using that failed for this use case.